File: Second Seminar/SecondSeminar/src/main/java/com/example/SecondSeminar/controller/dto/request/post/PostGetResponse.java
package com.example.SecondSeminar.controller.dto.request.post;

import com.example.SecondSeminar.controller.dto.response.category.CategoryResponse;
import com.example.SecondSeminar.domain.Category;
import com.example.SecondSeminar.domain.Post;

public record PostGetResponse(
        Long id,
        String title,
        String content,
        CategoryResponse category
) {

    public static PostGetResponse of(Post post, Category category) {
        return new PostGetResponse(
                post.getId(),
                post.getTitle(),
                post.getContent(),
                CategoryResponse.of(category)
        );
    }
}

File: Second Seminar/SecondSeminarKotlin/src/main/kotlin/com/example/SecondSeminarKotlin/post/dto/response/PostGetResponse.kt
package com.example.SecondSeminarKotlin.post.dto.response

import com.example.SecondSeminarKotlin.post.domain.Post

class PostGetResponse(
    val id: Long,
    val title: String,
    val content: String,
) {
    companion object {
        fun of(post: Post): PostGetResponse {
            return PostGetResponse(
                post.id,
                post.title,
                post.content,
            )
        }
    }
}